src/measure_theory/group.lean

```diff
@@ -43,73 +55,156 @@ namespace measure
+by { refine ⟨λ h, _, measure.regular.inv⟩, rw ←μ.inv_inv, exact measure.regular.inv h }
 
-lemma is_right_invariant_conj' (h : is_left_invariant μ) :
-  is_right_invariant μ.conj :=
+@[to_additive]
+lemma is_mul_right_invariant_inv' (h : is_mul_left_invariant μ) :
+  is_mul_right_invariant μ.inv :=
 begin
-  intros g A hA, rw [μ.conj_apply (measurable_mul_right g hA), μ.conj_apply hA],
+  intros g A hA, rw [μ.inv_apply (measurable_mul_right g hA), μ.inv_apply hA],
   convert h g⁻¹ (measurable_inv hA) using 2,
   simp only [←preimage_comp, ← inv_preimage],
   apply preimage_congr, intro h, simp only [mul_inv_rev, comp_app, inv_inv]
 end
 
-lemma is_left_invariant_conj' (h : is_right_invariant μ) : is_left_invariant μ.conj :=
+@[to_additive]
+lemma is_mul_left_invariant_inv' (h : is_mul_right_invariant μ) : is_mul_left_invariant μ.inv :=
 begin
-  intros g A hA, rw [μ.conj_apply (measurable_mul_left g hA), μ.conj_apply hA],
+  intros g A hA, rw [μ.inv_apply (measurable_mul_left g hA), μ.inv_apply hA],
   convert h g⁻¹ (measurable_inv hA) using 2,
   simp only [←preimage_comp, ← inv_preimage],
   apply preimage_congr, intro h, simp only [mul_inv_rev, comp_app, inv_inv]
 end
 
-@[simp] lemma is_right_invariant_conj : is_right_invariant μ.conj ↔ is_left_invariant μ :=
-by { refine ⟨λ h, _, is_right_invariant_conj'⟩, rw ←μ.conj_conj, exact is_left_invariant_conj' h }
+@[simp, to_additive]
+lemma is_mul_right_invariant_inv : is_mul_right_invariant μ.inv ↔ is_mul_left_invariant μ :=
+by { refine ⟨λ h, _, is_mul_right_invariant_inv'⟩, rw ← μ.inv_inv,
+     exact is_mul_left_invariant_inv' h }
+
+@[simp, to_additive]
+lemma is_mul_left_invariant_inv : is_mul_left_invariant μ.inv ↔ is_mul_right_invariant μ :=
+by { refine ⟨λ h, _, is_mul_left_invariant_inv'⟩, rw ← μ.inv_inv,
+     exact is_mul_right_invariant_inv' h }
+
+end inv
+
+namespace measure
+
+variables [measurable_space G] [topological_space G] [borel_space G] {μ : measure G}
+
+section group
+
+variables [group G] [topological_group G]
+
+/-! Properties of regular left invariant measures -/
+
+@[to_additive]
+lemma null_of_is_mul_left_invariant_of_ne_zero (hμ : μ.regular) (h2μ : is_mul_left_invariant μ)
+  (h3μ : μ ≠ 0) {s : set G} (hs : is_open s) : μ s = 0 ↔ s = ∅ :=
+begin
+  obtain ⟨K, hK, h2K⟩ := hμ.exists_compact_not_null.mpr h3μ,
+  refine ⟨λ h, _, λ h, by simp [h]⟩,
+  by_contradiction h2s, apply h2K,
+  rw [← ne.def, ne_empty_iff_nonempty] at h2s, cases h2s with y hy,
+  obtain ⟨t, -, h1t, h2t⟩ := hK.elim_finite_subcover_image
+    (show ∀ x ∈ @univ G, is_open ((λ y, x * y) ⁻¹' s),
+      from λ x _, (continuous_mul_left x).is_open_preimage _ hs) _,
+  { rw [← nonpos_iff_eq_zero],
+    refine (measure_mono h2t).trans _,
+    refine (measure_bUnion_le h1t.countable _).trans_eq _,
+    simp_rw [h2μ _ hs.is_measurable], rw [h, tsum_zero] },
+  { intros x _, simp_rw [mem_Union, mem_preimage], use [y * x⁻¹, mem_univ _],
```

Comment:
One tactic per line? Here and several times in the rest of the PR :-) (I promise, it's the last time I mention this).

Reply:
Feel free to keep mentioning this. I've already split a couple of tactic proofs into 1 tactic per line, but I will sometimes forget this rule in the foreseeable future.